pointgrey_camera_driver/src/nodelet.cpp

```diff
@@ -367,17 +367,62 @@ class PointGreyCameraNodelet: public nodelet::Nodelet
       }
       catch(std::runtime_error& e)
       {
+        // Something terrible has happened, so let's just disconnect and reconnect to see if we can recover.
         NODELET_ERROR("%s", e.what());
-        ///< @todo Look into readding this
-        /*try{
-          // Something terrible has happened, so let's just disconnect and reconnect to see if we can recover.
-          pg_.disconnect();
-          ros::Duration(1.0).sleep(); // sleep for one second each time
-          pg_.connect();
-          pg_.start();
-        } catch(std::runtime_error& e2){
-          NODELET_ERROR("%s", e2.what());
-        }*/
+
+        // Try disconnecting from the camera
```

Comment:
Is there some overlap here with the initial camera connection? Wondering if it can go to an inner-outer loop pattern, eg:

```
while (ok)
{
  connect()
  while (ok && connected)
  {
    grab_image()
    if (error) break
  }
  sleep(1)
  disconnect()
}
```